spec/rubocop/cop/style/collection_methods_spec.rb

```diff
@@ -17,14 +17,14 @@
   cop_config['PreferredMethods'].each do |method, preferred_method|
     it "registers an offense for #{method} with block" do
       inspect_source("[1, 2, 3].#{method} { |e| e + 1 }")
-      expect(cop.offenses.size).to eq(1)
```

Comment:
I removed this redundant checks, because the next expectation checks that array has exactly one element.

This improves test failure error messages as well.

Before:

```
1) RuboCop::Cop::Style::CollectionMethods registers an offense for collect with proc param
     Failure/Error: expect(cop.offenses.size).to eq(2)

       expected: 2
            got: 1

       (compared using ==)
     # ./spec/rubocop/cop/style/collection_methods_spec.rb:27:in `block (3 levels) in <top (required)>'
```

After: 
```
 1) RuboCop::Cop::Style::CollectionMethods registers an offense for collect with proc param
     Failure/Error:
       expect(cop.messages)
         .to eq([])

       expected: []
            got: ["Prefer `map` over `collect`."]

       (compared using ==)
     # ./spec/rubocop/cop/style/collection_methods_spec.rb:28:in `block (3 levels) in <top (required)>'
```